src/main/java/mobi/boilr/libdynticker/exchanges/BinanceExchange.java

```diff
@@ -0,0 +1,67 @@
+package mobi.boilr.libdynticker.exchanges;
+
+import mobi.boilr.libdynticker.core.Pair;
+import mobi.boilr.libdynticker.core.exception.NoMarketDataException;
+import org.codehaus.jackson.JsonNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+public class BinanceExchange extends mobi.boilr.libdynticker.core.Exchange {
+    public BinanceExchange(long expiredPeriod) {
+        super("Binance", expiredPeriod);
+    }
+
+    @Override
+    protected List<Pair> getPairsFromAPI() throws IOException {
+        JsonNode jsonNode = readJsonFromUrl("https://www.binance.com/api/v1/ticker/allPrices");
+        ArrayList<Pair> pairs = new ArrayList<>();
+        for (JsonNode node:jsonNode) {
+            String symbol = node.get("symbol").asText();
+            if(symbol.chars().allMatch(Character::isLetter)){
+                String coin = symbol.replaceFirst(Exchange.matchExchange(), "");
+                String exchange = symbol.replaceFirst(coin, "");
+                pairs.add(new Pair(coin, exchange));
+            }
+        }
+        return pairs;
+    }
+
+    @Override
+    protected String getTicker(Pair pair) throws IOException, NoMarketDataException {
+        JsonNode node = readJsonFromUrl("https://www.binance.com/api/v1/ticker/24hr?symbol=" + pair.getCoin() + pair.getExchange());
+        if(node.has("priceChange")) {
+            return parseTicker(node, pair);
+        }
+        throw new NoMarketDataException(pair);
+    }
+
+    @Override
+    public String parseTicker(JsonNode node, Pair pair) throws IOException {
+        return node.get("lastPrice").asText();
+    }
+
+    private enum Exchange {
```

Comment:
No need to programmatically build the regex. Please delete this enum and directly add the regex into a variable on `getPairsFromAPI`. This will shorten your code and make it more clear. Much better to have `(BTC|ETH|BNB|USDT)$` plainly written on `getPairsFromAPI` than to force people to mentally compile the regex from the 12 liner `matchExchange` method.